system-addon/test/unit/lib/FilterAdult.test.js

```diff
@@ -0,0 +1,56 @@
+const {GlobalOverrider} = require("test/unit/utils");
+
+describe("filterAdult", () => {
+  let globals;
+  let filterAdult;
+  let hashStub;
+  let hashValue;
+
+  beforeEach(() => {
+    globals = new GlobalOverrider();
+    globals.set("Services", {
+      eTLD: {getBaseDomain({spec}) { return spec.match(/\/([^/]+)/)[1]; }},
```

Comment:
You should be able to add ```getBaseDomain``` to ```unit-entry.js```'s overrider of Services. You can also alter the existing one for ```newURI``` to return ```{spec: url}``` like you have here and it won't fail any of the other tests

Reply:
Aha! indeed. No need for a separate `GlobalOverrider` here now.